lib/miq_expression/target.rb

```diff
@@ -9,11 +9,16 @@ def self.parse!(field)
   # returns hash:
   # {:model_name => Host<ApplicationRecord> , :associations => ['vms']<Array>, :column_name => 'host_name' <String>}
   def self.parse_params(field)
+    return unless field.kind_of?(String)
     match = self::REGEX.match(field) || return
     # convert matches to hash to format
     # {:model_name => 'User', :associations => ...}
     parsed_params = Hash[match.names.map(&:to_sym).zip(match.to_a[1..-1])]
-    parsed_params[:model_name] = parsed_params[:model_name].classify.safe_constantize
+    begin
+      parsed_params[:model_name] = parsed_params[:model_name].classify.safe_constantize
+    rescue LoadError # issues for case sensitivity (e.g.: VM vs vm)
+      parsed_params[:model_name] = nil
```

Comment:
I thought safe_constantize's purpose is to never raise a LoadError, but to return nil instead.  How would the LoadError path ever occur?

Reply:
@Fryguy 
```
"Vm".safe_constantize # good
nil.safe_constantize # good
"XYZ".safe_constantize # good
"VM".safe_constantize # error
```

Reply:
@Fryguy having trouble reproducing this

[safe_constantize](https://github.com/rails/rails/blob/5-1-stable/activesupport/lib/active_support/inflector/methods.rb#L311) does not catch `LoadError` - but I'm having trouble generating that error from a code base - maybe because it is from the auto_load path